doc/nrf/protocols/matter/end_product/watchdog.rst

```diff
@@ -0,0 +1,251 @@
+  If you want to keep low power consumption, consider using the manual feeding approach instead.
+
+  To configure and enable the automatic feeding feature, follow these steps:
+
+  1. Include the ``watchdog/watchdog.h`` file.
+
+     .. code-block:: c++
+
+        #include "watchdog/watchdog.h"
+
+  2. Declare and define the ``FeedingCallback`` implementation in your application code.
+     This implementation must be tailored to your needs.
+
+     For example, here is an implementation of ``FeedingCallback`` for feeding the Matter watchdog source within the Main thread:
+
+     .. code-block:: c++
+
+        void FeedFromApp(Nrf::Watchdog::WatchdogSource *source)
+        {
+	        if (source)
+            {
+		        Nrf::PostTask([source] { source->Feed(); });
+	        }
+        }
+
+  3. Create a Matter watchdog source and provide the ``uint32_t feedingInterval``, and ``FeedingCallback callback`` optional arguments to the Matter watchdog source constructor.
+
+     For example, here is a Matter watchdog source object configured to call the previously defined callback every 200 ms:
+
+     .. code-block:: c++
+
+        Nrf::Watchdog::WatchdogSource myWatchdog(200, FeedFromApp);
+
+  4. Try to install the Matter watchdog source and check the result of the function:
+
+     .. code-block:: c++
+
+        if(!Nrf::Watchdog::InstallSource(myWatchdog))
+        {
+            LOG_ERR("Watchdog source cannot be installed.");
+        }
+
+  5. Enable the Global Watchdog module and check the function result:
+
+     .. code-block:: c++
+
+        if(!Nrf::Watchdog::Enable())
+        {
+            return false;
+        }
+
+Feeding manually
+  Manual feeding involves calling the ``Feed()`` method of the specific source.
+  This approach provides the best power consumption, because feeding only takes place if the CPU is not in sleep mode.
+
+  To create, enable, and manually feed the Matter Watchdog source, complete the following steps:
+
+  1. Create a Matter Watchdog source without any arguments.
+
+     For example:
+
+     .. code-block:: c++
+
+        Nrf::Watchdog::WatchdogSource myWatchdog;
+
+  2. Try to install the Matter watchdog source and check the result of the function:
+
+     .. code-block:: c++
+
+        if(!Nrf::Watchdog::InstallSource(myWatchdog))
+        {
+            LOG_ERR("Watchdog source cannot be installed.");
+        }
+
+  3. Enable the Global Watchdog module and check the function result:
+
+    .. code-block:: c++
+
+        if(!Nrf::Watchdog::Enable())
+        {
+            return false;
+        }
+
+  4. Call the ``Feed()`` method at the specific place in the code where you want to prevent code blocking:
+
+     .. code-block:: c++
+
+        myWatchdog.Feed();
+
+
+.. note::
+
+    If the ``InstallSource(WatchdogSource &source)`` function returns ``false`` it may mean that there is no available watchdog channel to assign.
```

Comment:
```suggestion
    If the ``InstallSource(WatchdogSource &source)`` function returns ``false``, it can mean that there is no available watchdog channel to assign.
```